lib/media/drm_engine.js

```diff
@@ -1012,8 +1032,14 @@ shaka.media.DrmEngine = class {
    */
   createTemporarySession_(initDataType, initData) {
     let session;
+
+    const mediaKeySystemAccessConfiguration =
+      this.mediaKeySystemAccess_.getConfiguration();
+
     try {
-      if (this.usePersistentLicenses_) {
+      if (this.usePersistentLicenses_ ||
+        mediaKeySystemAccessConfiguration.persistentState === 'required'
```

Comment:
As above, we can still use persistent state with `temporary` licenses.  There should be a different config for the session type to use.  Maybe another field in `DrmInfo`.